shared/utils/TasksProvider/TasksProvider.tsx

```diff
@@ -82,33 +77,27 @@ function TasksProvider({ children }: IProps) {
     }
   };
 
-  const filterByPriorty = (priority: string) => {
-    const filteredTasks = data?.filter(
-      (task: TTask) => task.priority === priority
-    );
-    setFilter({ by: "priority", value: priority });
-    setTasks(filteredTasks);
-    return filteredTasks;
-  };
+  const filterTasks = (filter: TFilter) => {
+    const filteredTasks = data?.filter((task) => {
+      const priorityMatch = filter?.priority
+        ? task.priority === filter.priority
+        : true;
```

Comment:
```suggestion
      const priorityMatch = !filter?.priority || task.priority === filter.priority;
```

[NIT]: Found this logic a bit confusing. Can do something like this to slightly simplify and make it more readable in my opinion. Same comment for similar logic structures in this PR.

P.S. NIT comments are comments you can choose to ignore. It's up to the author (you) whether they want to make that change or not.